### PR TITLE
Prevent adding hosts that conflict with CNAME records.

### PR DESCRIPTION
**Pull request by mbakke**
See #417 for background.

### CODE REVIEW

**Comment by mbakke** [comment]
That's a great point, confirmed by a new test:

```
    def test_hosts_patch_409_conflict_cname(self):
        """Patching an entry with a CNAME that already exists should return 409"""
        post_data_cname = {'name': 'new-name2.example.org',
                           'host': self.host_two.id}
        self.assert_post('/cnames/', post_data_cname)
        self.assert_patch_and_409('/hosts/%s' % self.host_one.name,
                                  {'name': 'new-name2.example.org'})
```

I suppose the CNAME check should be done in serializer, will investigate how to raise a specific HTTP status code as discussed in #417.

**Comment by mbakke** [comment]
@oyvindkolbu I've pushed a v2 that adds a new `ValidationError409` class and raises it in the serializer on CNAME conflicts.

Not sure whether `serializers.py` is a good home for that `ValidationError409` class though.

**Comment by mbakke** [comment]
v3 of this patch introduces `mreg.errors` for the custom `ValidationError409` class.

**Review by oyvindhagberg — APPROVED** [review]

File: mreg/errors.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from rest_framework import (exceptions, status)
+
+class ValidationError409(exceptions.APIException):
+    status_code = status.HTTP_409_CONFLICT
```
Comment (oyvindkolbu) [inline mreg/errors.py:4]: Except that this file should probably reside under `mreg/api/` or `mreg/api/v1/` it looks good.

Comment (mbakke) [reply mreg/errors.py:4]: That makes sense.  Moved to `mreg.api.errors`.  Thanks a lot for the feedback!  :-)

**Review by oyvindhagberg — APPROVED** [review]

**Pull request merged**